grouper/ctl/user_proxy.py

```diff
@@ -1,79 +1,147 @@
 from __future__ import print_function
 
-import BaseHTTPServer
 import getpass
 import logging
-import urllib2
+from typing import cast, TYPE_CHECKING
 
-from mrproxy import UserProxyHandler
+from grouper.ctl.base import CtlCommand
 
+try:
+    from http.server import BaseHTTPRequestHandler, HTTPServer
+    from urllib.error import HTTPError, URLError
+    from urllib.request import build_opener, HTTPErrorProcessor, Request
+except ImportError:
```

Comment:
Out of curiosity -- is this a hack to make it work with both Python2 and 3?

Reply:
Yup! Although now that I look at this again in the morning, I'm going to switch to using `six.moves`, which should make this more obvious and not require as much of an awkward workaround.